firebase-sessions/src/main/kotlin/com/google/firebase/sessions/ProcessDataManager.kt

```diff
@@ -81,17 +103,22 @@ constructor(private val appContext: Context, private val uuidGenerator: UuidGene
       ?.toMap()
       ?: mapOf(myProcessName to ProcessData(Process.myPid(), myUuid))
 
+  /** Gets the current details for all of the app's running processes. */
+  private fun getAppProcessDetails() = ProcessDetailsProvider.getAppProcessDetails(appContext)
+
   /**
    * Returns true if the process is stale, meaning the persisted process data does not match the
    * running process details.
+   *
+   * The [processDetails] is the running process details, and [processData] is the persisted data.
    */
-  private fun isProcessStale(
-    runningProcessDetails: ProcessDetails,
-    persistedProcessData: ProcessData,
-  ): Boolean =
-    if (myProcessName == runningProcessDetails.processName) {
-      runningProcessDetails.pid != persistedProcessData.pid || myUuid != persistedProcessData.uuid
+  private fun isProcessStale(processDetails: ProcessDetails, processData: ProcessData): Boolean =
+    if (myProcessName == processDetails.processName) {
+      // For this process, check pid and uuid
+      processDetails.pid != processData.pid || myUuid != processData.uuid
```

Comment:
Potentially change this to call `isMyProcessStale`.

Reply:
This function is different, it is general so works on any process. If it delegated this case to `isMyProcessStale`, it wouldn't look at the provided processDetails, it would look at `myProcessDetails` instead.